src/ui/guiforcase4/AddVehicles.java

```diff
@@ -65,51 +83,135 @@ private void addComponents(JFrame frame, JPanel panel) {
+        model.setModel(new DefaultComboBoxModel(models1));
+        model.setBounds(160, 140, 160, 25);
+        make.addItemListener(new ItemListener() {
+            @Override
+            public void itemStateChanged(ItemEvent e) {
+                if (e.getStateChange() == ItemEvent.SELECTED) {
+                    String makeValue = make.getSelectedItem().toString();
+                    Collection<String> models = makeModel(makeValue).getModels();
+                    String[] models1 = models.toArray(new String[models.size()]);
+                    model.setModel(new DefaultComboBoxModel(models1));
+                }
+
+            }
+        });
+
+        panel.add(make);
+        panel.add(model);
+
+        btn1.addActionListener(new ActionListener() {
             @Override
             public void actionPerformed(ActionEvent e) {
-                new InventoryInformation(dID);
-                frame.dispose();
+                vehicle.setVin(Integer.parseInt(tf2.getText()));
+                vehicle.setDealerId(dID);
+                vehicle.setMake(make.getSelectedItem().toString());
+                vehicle.setModel(model.getSelectedItem().toString());
+                vehicle.setYear(Integer.parseInt(cmb3.getSelectedItem().toString()));
+                vehicle.setCategory(cmb2.getSelectedItem().toString());
+                vehicle.setPrice(Float.parseFloat(tf3.getText()));
+                vehicle.setColor(cmb1.getSelectedItem().toString());
+                vehicle.setMileage(Integer.parseInt(tf4.getText()));
+                vmi.addVehicle(vehicle);
             }
         });
+
+
+    }
+    private MakeModel makeModel(String makeValue) {
+        makeModels =(ArrayList<MakeModel>) (mmc.getMakeModels());
+        for (MakeModel mm : makeModels) {
+            if (mm.getMake().equals(makeValue)) {
+                return mm;
+            }
+        }
+        return null;
+    }
+
+    public String[] getYearsFromDatabase() {
+        ArrayList<Vehicle>  vehicles = connect.executeVehicleQuery();
+        String[] years = new String[vehicles.size()];
+        for (int i = 0; i < vehicles.size(); i++) {
+            years[i] = vehicles.get(i).getYear()+"";
+        }
+        return years;
+    }
+    public String[] getColorsFromDatabase() {
+        ArrayList<Vehicle> vehicles = connect.executeVehicleQuery();
+        String[] color = new String[vehicles.size()];
+        for (int i = 0; i < vehicles.size(); i++) {
+            color[i] = vehicles.get(i).getColor();
+        }
+        return color;
+    }
+    public String[] getMakesFromDatabase() {
+        ArrayList<Vehicle> vehicles = connect.executeVehicleQuery();
+        String[] makes = new String[vehicles.size()];
+        for (int i = 0; i < vehicles.size(); i++) {
+            makes[i] = vehicles.get(i).getMake();
+        }
+        return makes;
+    }
+
+    public String[] removeDuplicates(String[] array) {
```

Comment:
The idea is good, but I was wondering if there is any way to check for duplicates within the SQL query itself?

Reply:
We will be refactoring this part in upcoming version

Reply:
You are right, Lakshya! Of Course, we can use SQL query to fetch distinct values of attributes. This query "SELECT DISTINCT column1, column2, column3" can only get distinct values of column1, I was too lazy to separate them into three methods, so...    I will try to improve them later. Thank you~